### PR TITLE
🐛 Fixes several issues in director-v2 integration test

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ``pytest_simcore.docker_swarm.docker_stack`` fixture would wait endlessly to networks to be down during tear-down, just after the ``docker swarm rm`` call.  The latter reported an error as we
   - Added a limited retry
   - Fixes logs in 
   - logs problematic resources
- redirection to director is faulty:
   - redirection to director ends in director-v2 /run_interactive which had different query entry ``service_version`` -> ``service_tag``
   - httpx client has problems to redirect in the test  
- merges changes from PR #2623

### CODE REVIEW

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 1]
```diff
@@ -205,41 +214,44 @@ def docker_stack(
...
     for _, stack, _ in stacks:
 
         try:
-            subprocess.run(f"docker stack remove {stack}", shell=True, check=True)
+            subprocess.run(
+                f"docker stack remove {stack}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:228]: I would always capture the output from here, and always print it. Even if no error occurs.

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 2]
```diff
@@ -25,6 +26,14 @@
 
 log = logging.getLogger(__name__)
 
+_MINUTES = 60  # secs
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:29]:
```suggestion
_MINUTE = 60  # secs

```
sounds better

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 3]
```diff
@@ -205,41 +214,44 @@ def docker_stack(
...
+            resource = getattr(docker_client, resource_name)
+
+            for attempt in Retrying(
+                wait=wait_exponential(), stop=stop_after_delay(3 * _MINUTES)
+            ):
+                with attempt:
+                    if pending := resource.list(
+                        filters={"label": f"com.docker.stack.namespace={stack}"}
+                    ):
+                        msg = f"Waiting for {len(pending)} {resource_name} to shutdown [{pending[:3]} ... ]."
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:250]: 👍 could you also add something like this for the volumes below?

**Review by GitHK — APPROVED** [review]

**Review by odeimaiz — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py [hunk 4]
```diff
@@ -376,12 +383,16 @@ async def shutdown(self):
         self._to_observe = {}
 
         if self._scheduler_task is not None:
-            await self._scheduler_task
+            self._scheduler_task.cancel()
+            with contextlib.suppress(asyncio.CancelledError):
+                await self._scheduler_task
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py:388]: I see this was missing

File: services/director-v2/tests/integration/02/test_mixed_dynamic_sidecar_and_legacy_project.py [hunk 5]
```diff
@@ -220,13 +215,13 @@ async def test_legacy_and_dynamic_sidecar_run(
     - dy-static-file-server-dynamic-sidecar
     - dy-static-file-server-dynamic-sidecar-compose
     """
-    director_v0_url = get_director_v0_patched_url(services_endpoint["director"])
+    # FIXME: ANE can you instead parametrize this test?
+    # why do we need to run all these services at the same time? it would be simpler one by one
```
Comment (GitHK) [inline services/director-v2/tests/integration/02/test_mixed_dynamic_sidecar_and_legacy_project.py:219]: The idea behind this was to test that these services could all run in parallel. Why would you change it to only one service?

File: services/director-v2/tests/integration/02/utils.py [hunk 6]
```diff
@@ -198,45 +150,46 @@ async def assert_start_service(
...
     )
-    result = await handle_307_if_required(director_v2_client, director_v0_url, result)
-    assert result.status_code == 200, result.text
-
-    payload = result.json()
+    if response.status_code == httpx.codes.TEMPORARY_REDIRECT:
+        # NOTE: so we have a redirect, and it seems the director_v2_client does not like it at all
+        #  moving from the testserver to the director in this GET call
+        # which is why we use a DIFFERENT httpx client for this... (sic).
+        # This actually works well when running inside the swarm... WTF???
```
Comment (GitHK) [inline services/director-v2/tests/integration/02/utils.py:172]: Not sure about what you mean on this last line.

Comment (sanderegg) [reply services/director-v2/tests/integration/02/utils.py:172]: yes this comes from your comments.

File: services/director-v2/tests/integration/02/test_mixed_dynamic_sidecar_and_legacy_project.py (hunk 5 above)
Comment (sanderegg) [reply services/director-v2/tests/integration/02/test_mixed_dynamic_sidecar_and_legacy_project.py:219]: to simplify debugging

**Pull request merged**